beacon-chain/operations/slashings/service_new.go

```diff
@@ -48,37 +48,34 @@ func (p *PoolService) run() {
 		return
 	}
 
-	electraSlot, err := slots.EpochStart(params.BeaconConfig().ElectraForkEpoch)
-	if err != nil {
-		log.WithError(err).Error("Could not get Electra start slot")
-		return
-	}
-
 	// If run() is executed after the transition to Electra has already happened,
 	// there is nothing to convert because the slashing pool is empty at startup.
-	if p.currentSlotFn() >= electraSlot {
+	if slots.ToEpoch(p.currentSlotFn()) >= params.BeaconConfig().ElectraForkEpoch {
 		return
 	}
 
 	p.waitForChainInitialization()
 
-	electraTime, err := slots.ToTime(uint64(p.clock.GenesisTime().Unix()), electraSlot)
-	if err != nil {
-		log.WithError(err).Error("Could not get Electra start time")
-		return
-	}
-
-	t := time.NewTimer(electraTime.Sub(p.clock.Now()))
-	defer t.Stop()
-
-	select {
-	case <-t.C:
-		log.Info("Converting Phase0 slashings to Electra slashings")
-		p.poolManager.ConvertToElectra()
-		return
-	case <-p.ctx.Done():
-		log.Warn("Context cancelled, ConvertToElectra timer will not execute")
-		return
+	ticker := time.NewTicker(time.Duration(params.BeaconConfig().SecondsPerSlot) * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-p.ctx.Done():
+			log.Warning("Context cancelled, ConvertToElectra aborted")
+			return
+		default:
+			if slots.ToEpoch(p.currentSlotFn()) >= params.BeaconConfig().ElectraForkEpoch {
+				log.Info("Converting Phase0 slashings to Electra slashings")
+				p.poolManager.ConvertToElectra()
+				return
+			}
+		}
+		select {
```

Comment:
Why do you have a second `select`?

Reply:
The first one checks for the context being done, the second for the ticker to trigger.

Reply:
But I can do it with one too